Description: We install all scripts and programs of TigerVNC with a tiger prefix.
 Hence, we have to adapt some programs to take this into account.
Author: Joachim Falk <joachim.falk@gmx.de>

Index: pkg-tigervnc/unix/vncserver/vncsession.c
===================================================================
--- pkg-tigervnc.orig/unix/vncserver/vncsession.c
+++ pkg-tigervnc/unix/vncserver/vncsession.c
@@ -417,7 +417,7 @@ run_script(const char *username, const c
 {
     struct passwd *pwent;
     pid_t pid;
-    const char *child_argv[3];
+    const char *child_argv[4];
 
     pwent = getpwnam(username);
     if (pwent == NULL) {
@@ -470,9 +470,10 @@ run_script(const char *username, const c
     setenv("USER", pwent->pw_name, 1);
     setenv("USERNAME", pwent->pw_name, 1);
 
-    child_argv[0] = CMAKE_INSTALL_FULL_LIBEXECDIR "/vncserver";
-    child_argv[1] = display;
-    child_argv[2] = NULL;
+    child_argv[0] = CMAKE_INSTALL_FULL_BINDIR "/tigervncserver";
+    child_argv[1] = "-fg";
+    child_argv[2] = display;
+    child_argv[3] = NULL;
 
     execvp(child_argv[0], (char*const*)child_argv);
 
@@ -516,7 +517,7 @@ main(int argc, char **argv)
     if (begin_daemon() == -1)
         return EX_OSERR;
 
-    openlog("vncsession", LOG_PID, LOG_AUTH);
+    openlog("tigervncsession", LOG_PID, LOG_AUTH);
 
     /* Indicate that this is a graphical user session. We need to do
        this here before PAM as pam_systemd.so looks at these. */
@@ -549,13 +550,13 @@ main(int argc, char **argv)
 
     script = run_script(username, display, child_env);
     if (script == -1) {
-        syslog(LOG_CRIT, "Failure starting vncserver script");
+        syslog(LOG_CRIT, "Failure starting tigervncserver script");
         stop_pam(pamh, pamret);
         return EX_OSERR;
     }
 
     snprintf(pid_file, sizeof(pid_file),
-             "/run/vncsession-%s.pid", display);
+             "/run/tigervncsession-%s.pid", display);
     f = fopen(pid_file, "w");
     if (f == NULL) {
         syslog(LOG_ERR, "Failure creating pid file \"%s\": %s",
@@ -580,14 +581,14 @@ main(int argc, char **argv)
         if (WIFEXITED(status)) {
             if (WEXITSTATUS(status) != 0) {
                 syslog(LOG_WARNING,
-                        "vncsession: vncserver exited with status=%d",
+                        "tigervncsession: tigervncserver exited with status=%d",
                         WEXITSTATUS(status));
             }
             break;
         }
         else if (WIFSIGNALED(status)) {
             syslog(LOG_WARNING,
-                    "vncsession: vncserver was terminated by signal %d",
+                    "tigervncsession: tigervncserver was terminated by signal %d",
                     WTERMSIG(status));
             break;
         }
Index: pkg-tigervnc/unix/vncserver/vncserver@.service.in
===================================================================
--- pkg-tigervnc.orig/unix/vncserver/vncserver@.service.in
+++ pkg-tigervnc/unix/vncserver/vncserver@.service.in
@@ -1,11 +1,11 @@
-# The vncserver service unit file
+# The tigervncserver service unit file
 #
 # Quick HowTo:
 # 1. Add a user mapping to @CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver.users.
 # 2. Adjust the global or user configuration. See the
-#    vncsession(8) manpage for details. (OPTIONAL)
-# 3. Run `systemctl enable vncserver@:<display>.service`
-# 4. Run `systemctl start vncserver@:<display>.service`
+#    tigervncsession(8) manpage for details. (OPTIONAL)
+# 3. Run `systemctl enable tigervncserver@:<display>.service`
+# 4. Run `systemctl start tigervncserver@:<display>.service`
 #
 # DO NOT RUN THIS SERVICE if your local area network is
 # untrusted!  For a secure way of using VNC, you should
@@ -35,8 +35,8 @@ After=syslog.target network.target
 
 [Service]
 Type=forking
-ExecStart=@CMAKE_INSTALL_FULL_LIBEXECDIR@/vncsession-start %i
-PIDFile=/run/vncsession-%i.pid
+ExecStart=@CMAKE_INSTALL_FULL_LIBEXECDIR@/tigervncsession-start %i
+PIDFile=/run/tigervncsession-%i.pid
 SELinuxContext=system_u:system_r:vnc_session_t:s0
 
 [Install]
Index: pkg-tigervnc/unix/vncserver/vncsession-start.in
===================================================================
--- pkg-tigervnc.orig/unix/vncserver/vncsession-start.in
+++ pkg-tigervnc/unix/vncserver/vncsession-start.in
@@ -40,4 +40,4 @@ if [ -z "${USER}" ]; then
 	exit 1
 fi
 
-exec "@CMAKE_INSTALL_FULL_SBINDIR@/vncsession" "${USER}" "${DISPLAY}"
+exec "@CMAKE_INSTALL_FULL_SBINDIR@/tigervncsession" "${USER}" "${DISPLAY}"
Index: pkg-tigervnc/unix/vncserver/vncsession.man.in
===================================================================
--- pkg-tigervnc.orig/unix/vncserver/vncsession.man.in
+++ pkg-tigervnc/unix/vncserver/vncsession.man.in
@@ -17,44 +17,48 @@ is rarely called directly and is normall
 manager.
 
 .SH FILES
-Several VNC-related files are found in the directory $HOME/.vnc:
+Several VNC-related files are found in the directory \fI~/.vnc\fP:
 .TP
-@CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver-config-defaults
-The optional system-wide equivalent of $HOME/.vnc/config. If this file exists
+.I @CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver-config-defaults
+The optional system-wide equivalent of \fI~/.vnc/tigervnc.conf\fP. If this file exists
 and defines options to be passed to Xvnc, they will be used as defaults for
-users. The user's $HOME/.vnc/config overrides settings configured in this file.
-The overall configuration file load order is: this file, $HOME/.vnc/config,
-and then @CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver-config-mandatory. None are required to exist.
+users. The user's \fI~/.vnc/tigervnc.conf\fP overrides settings configured in this file.
+The overall configuration file load order is: this file, \fI~/.vnc/tigervnc.conf\fP,
+and then \fI@CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver-config-mandatory\fP. None are required to exist.
 .TP
-@CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver-config-mandatory
-The optional system-wide equivalent of $HOME/.vnc/config. If this file exists
+.I @CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver-config-mandatory
+The optional system-wide equivalent of \fI~/.vnc/tigervnc.conf\fP. If this file exists
 and defines options to be passed to Xvnc, they will override any of the same
-options defined in a user's $HOME/.vnc/config. This file offers a mechanism
+options defined in a user's \fI~/.vnc/tigervnc.conf\fP. This file offers a mechanism
 to establish some basic form of system-wide policy. WARNING! There is
 nothing stopping users from constructing their own vncsession-like script
 that calls Xvnc directly to bypass any options defined in
-@CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver-config-mandatory. The overall configuration file load
-order is: @CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver-config-defaults, $HOME/.vnc/config, and then
+\fI@CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver-config-mandatory\fP. The overall configuration file load
+order is: \fI@CMAKE_INSTALL_FULL_SYSCONFDIR@/tigervnc/vncserver-config-defaults\fP, \fI~/.vnc/tigervnc.conf\fP, and then
 this file. None are required to exist.
 .TP
-$HOME/.vnc/config
+.I ~/.vnc/tigervnc.conf
 An optional server config file wherein options to be passed to Xvnc are listed
-to avoid hard-coding them to the physical invocation. List options in this file
-one per line. For those requiring an argument, simply separate the option from
-the argument with an equal sign, for example: "geometry=2000x1200" or
-"securitytypes=vncauth,tlsvnc". Options without an argument are simply listed
-as a single word, for example: "localhost" or "alwaysshared".
-
-The special option
+to avoid hard-coding them to the physical invocation. This file uses
+.BR perl (1)
+syntax. The special option
 .B session
 can be used to control which session type will be started. This should match
-one of the files in \fI/usr/share/xsessions\fP. E.g. if there is a file called
-"gnome.desktop", then "session=gnome" would be set to use that session type.
+one of the files in \fI/usr/share/xsessions\fP. E.g., if there is a file called
+\fIgnome.desktop\fP, then \fC$session="gnome";\fP would use that session type.
+
+To be compatible with the upstream provided wrapper scripts, we will fall back
+to trying to load configuration from \fI~/.vnc/config\fP if \fItigervnc.conf\fP is
+not present. The \fI~/.vnc/config\fP file lists options one per line. Options
+without an argument are listed as a single word, for example: "localhost" or
+"alwaysshared". For those requiring an argument, separate the option from the
+argument with an equal sign, for example, "session=gnome", "geometry=2000x1200",
+or "securitytypes=VncAuth,TLSVnc".
 .TP
-$HOME/.vnc/passwd
+~/.vnc/passwd
 The VNC password file.
 .TP
-$HOME/.vnc/\fIhost\fP:\fIdisplay#\fP.log
+~/.vnc/\fIhost\fP:\fIdisplay#\fP.log
 The log file for Xvnc and the session.
 
 .SH SEE ALSO
